esrecurse.js

```diff
@@ -53,8 +55,13 @@
         return (nodeType === estraverse.Syntax.ObjectExpression || nodeType === estraverse.Syntax.ObjectPattern) && key === 'properties';
     }
 
-    function Visitor(visitor) {
+    function Visitor(visitor, options) {
+        options = options || {};
+
         this.__visitor = visitor ||  this;
+        this.__childVisitorKeys = options.childVisitorKeys
```

Comment:
There is no need in those hacks. `estraverse` supports `keys` option for custom keys, where you can just pass this object from `esrecurse` as-is.